src/libtomahawk/playlist/TrackDetailView.cpp

```diff
@@ -441,3 +441,11 @@ TrackDetailView::setBuyButtonVisible( bool visible )
 {
     m_buyButtonVisible = visible;
 }
+
+
+void
+TrackDetailView::onResultClicked( const Tomahawk::playlistinterface_ptr& playlist, const Tomahawk::result_ptr& result, const Tomahawk::query_ptr& fromQuery )
+{
+    fromQuery->setPreferredResult( result );
```

Comment:
Yeah, that would be ok, but with your change it - it'll start playing even if the other query was playing, right?